code/modules/modular_computers/computers/item/computer_damage.dm

```diff
@@ -40,7 +40,7 @@
 	var/turf/newloc = get_turf(src)
 	new /obj/item/stack/sheet/metal(newloc, round(steel_sheet_cost/2))
 	for(var/C in all_components)
-		var/obj/item/weapon/computer_hardware/H = C
+		var/obj/item/weapon/computer_hardware/H = all_components[C]
 		uninstall_component(H)
```

Comment:
in case anyone is wondering, it's totally fine to modify the list while iterating over it with for!!

byond appears to create a local copy and iterates over that instead.